### DISCUSSION TITLE
Player Function (simple parsing shortcut)

### DISCUSSION THREAD

**Discussion by ShaneBeee** [SkriptLang/Skript#4007] [Suggestions]
This discussion is to suggest the introduction of a player function, which basically could be used as a shortcut for parsing.
Simply enter a player's name or uuid, and it will return a player object (or possible an offline player object if said player is not online)

ie:
```vb
send "blah" to player("ShaneBee")
```
or
```vb
send "blah" to player("some-uuid-here")
```

**Comment by Itzdlg** [comment]
In my opinion, I think that this is nice but unnecessary. If player's want this, they could add the function themselves. If we added this, then adding functions for every type would make more sense. It's also useless because Pyon said we might be removing functions, and just having custom syntax. It's also just as much typing as `"ShaneBee" as player`, so it doesn't change much.

**Reply by AyhamAl-Ali** [reply]
Removing functions?? I haven't heard about this but my first impression it is a very bad idea! many users depend on functions A LOT, it is just like methods unless there is another explanation for what you/snow said.

**Reply by FranKusmiruk** [reply]
Even though we are indeed planning on **replacing** functions at some point, it isn't not going to happen anytime soon and considering the amount of users that try to parse player names or uuids, I think it is adequate to have this utility function. That being said, having the `parsed` part of the parse expression be optional was also considered as an option and I am not against it, I'll let users decide through votes what they think is the best choice.

**Reply by AyhamAl-Ali** [reply]
Just wondering why replacing functions? and what will be the replacement?

**Reply by TheLimeGlass** [reply]
Not replacing functions as a whole. Replacing the default functions.

**Reply by TPGamesNL** [reply]
No, its about replacing functions as a whole, or at least remake their syntax

**Comment by AyhamAl-Ali** [comment]
This can be useful for many, I myself use `parsing` a lot on players and it don't like it, this will replace this long syntax `%(string|uuid)% parsed as [offline]player` which will save some time and effort when used a lot.

**Comment by TheLimeGlass** [comment]
The function is very ambiguous since there is offlineplayers. What happens if the player isn't online? It just returns nothing? Will you need to have offlinePlayer("Notch")? I think `"ShaneBee" as player` is a good addition. You could even go as far as removing the word `parsed` in the parse expression and just have shorthand `as`. That would be simplistic and I like that alot more now that I think about it.

**Comment by bluelhf** [comment]
I think this is a good idea, but since the current plan is to slowly replace functions as a concept (with reflect-like custom syntax I believe), I also think it'd be better to have syntax like `player "Notch"` or `player "4aa08588-1d1e-4d75-bd50-433c95ea9fb3"`, or even `entity "7fce36f1-1e14-4203-b8c9-ed44b74c5752"`

**Reply by AyhamAl-Ali** [reply]
I like the idea of player and entity expression `(player|entity) [from] %string%`
